Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -2948,20 +2949,28 @@ - (void)sendMessageAttachment:(SignalAttachment *)attachment
     }
 }
 
-- (void)sendContactShare:(OWSContact *)contactShare
+- (void)sendContactShare:(ContactShareViewModel *)contactShare
 {
     OWSAssertIsOnMainThread();
     OWSAssert(contactShare);
 
     DDLogVerbose(@"%@ Sending contact share.", self.logTag);
 
     BOOL didAddToProfileWhitelist = [ThreadUtil addThreadToProfileWhitelistIfEmptyContactThread:self.thread];
-    TSOutgoingMessage *message = [ThreadUtil sendMessageWithContactShare:contactShare
-                                                                inThread:self.thread
-                                                           messageSender:self.messageSender
-                                                              completion:nil];
 
-    [self messageWasSent:message];
+    [self.editingDatabaseConnection asyncReadWriteWithBlock:^(YapDatabaseReadWriteTransaction *_Nonnull transaction) {
```

Comment:
If this is going to be an async transaction, the `ensureDynamicInteractions` work below should be moved into the completion block.

Reply:
Done.